cloudvolume/datasource/graphene/mesh/sharded.py

```diff
@@ -188,7 +189,9 @@ def get_meshes_via_manifest_byte_offsets(self, seg_id, bounding_box):
     return dynamic_meshes + initial_meshes
 
   def get_meshes_via_manifest_labels(self, seg_id, bounding_box):
-    level = self.meta.meta.decode_layer_id(seg_id)
+    level = None
```

Comment:
I don't think I understand what this is doing here. What levels are specified if there is no bounding box? Does the server decide arbitrarily? Was it wrong to specify the level beforehand?

Reply:
The levels parameter is an optional one in the api, it means that the PCG should only return nodeIDs starting at this level in the oct-tree hierarchy.  The old code was saying that it should start with fragments that start at the level of the requested ID.  The problem is that the meshing might only be stitched up to a certain level, like level 7 for minnie. So when you combine that with speculative meshes, you are telling the api that you don't want it to check if a mesh exists and you want meshes at the level of the seg_id you are requesting (which for root_id's will be level 12), and you don't want it to check whether the mesh exists, just send you the path based upon the timestamp of the node.  

The old implementation was actually ignoring the level parameter if it was passed via speculative meshes.  This breaks a different function, when you actually do want to specify level=2 when you are doing bounding box queries because you don't want it returning fragments from higher up the hierarchy as the bounding box calculation is going to be messy in such a case.

Reply:
the new implementation fixes this and pays attention to the level parameter, but now the old cloud-volume code causes a bug where it returns a pointer to the level 12 nodeID, which then cloud volume discovers does not exist, silently ignores that a fragment doesn't exist, and then errors because its list of meshes is empty.  

The server is the entity that knows what level meshes are stitched up to, so really not passing level is the correct thing to do for speculative meshes.

Reply:
Gottcha. Sorry for taking so long to get to this. That one line was causing me anxiety and I should have taken the time to ask before.